### PR TITLE
fix for #3161

### PR DESCRIPTION
**Pull request by schnittchen**

### CODE REVIEW

File: lib/elixir/test/elixir/module_test.exs [hunk 1]
```diff
@@ -164,6 +164,12 @@ defmodule ModuleTest do
     module = Very.Long.Module.Name.And.Even.Longer
     assert Module.split(module) == ["Very", "Long", "Module", "Name", "And", "Even", "Longer"]
     assert Module.split("Elixir.Very.Long") == ["Very", "Long"]
+    assert_raise FunctionClauseError, fn ->
+      Module.split(:just_an_atom) # #3161 regression
```
Comment (josevalim) [inline lib/elixir/test/elixir/module_test.exs]: We can drop the comments here, just remember to mention it in the commit message. :)

**Comment by josevalim** [comment]
I have added a tiny comment, can you please amend and squash your changes into one? Thank you!

**Comment by jcspencer** [comment]
Should it be documented that `Module.split` interprets input this way?

**Comment by josevalim** [comment]
Improve the docs is a great idea, yes.

**Comment by schnittchen** [comment]
I wouldn't even mention strings... But add "The passed atom has to be a valid Elixir module name."

**Comment by josevalim** [comment]
We should definitely mention both. Both are officially accepted.

**Comment by schnittchen** [comment]
I've added 7194661c to my fork.

**Pull request merged**